CMakeLists.txt

```diff
@@ -0,0 +1,1131 @@
+  libc-top-half/musl/src/complex/cexp.c
+  libc-top-half/musl/src/complex/__cexpf.c
+  libc-top-half/musl/src/complex/cexpf.c
+  libc-top-half/musl/src/complex/cexpl.c
+  libc-top-half/musl/src/complex/cimagl.c
+  libc-top-half/musl/src/complex/clog.c
+  libc-top-half/musl/src/complex/clogf.c
+  libc-top-half/musl/src/complex/clogl.c
+  libc-top-half/musl/src/complex/conj.c
+  libc-top-half/musl/src/complex/conjf.c
+  libc-top-half/musl/src/complex/conjl.c
+  libc-top-half/musl/src/complex/cpow.c
+  libc-top-half/musl/src/complex/cpowf.c
+  libc-top-half/musl/src/complex/cpowl.c
+  libc-top-half/musl/src/complex/cproj.c
+  libc-top-half/musl/src/complex/cprojf.c
+  libc-top-half/musl/src/complex/cprojl.c
+  libc-top-half/musl/src/complex/creall.c
+  libc-top-half/musl/src/complex/csin.c
+  libc-top-half/musl/src/complex/csinf.c
+  libc-top-half/musl/src/complex/csinh.c
+  libc-top-half/musl/src/complex/csinhf.c
+  libc-top-half/musl/src/complex/csinhl.c
+  libc-top-half/musl/src/complex/csinl.c
+  libc-top-half/musl/src/complex/csqrt.c
+  libc-top-half/musl/src/complex/csqrtf.c
+  libc-top-half/musl/src/complex/csqrtl.c
+  libc-top-half/musl/src/complex/ctan.c
+  libc-top-half/musl/src/complex/ctanf.c
+  libc-top-half/musl/src/complex/ctanh.c
+  libc-top-half/musl/src/complex/ctanhf.c
+  libc-top-half/musl/src/complex/ctanhl.c
+  libc-top-half/musl/src/complex/ctanl.c
+  libc-top-half/musl/src/crypt/crypt_blowfish.c
+  libc-top-half/musl/src/crypt/crypt.c
+  libc-top-half/musl/src/crypt/crypt_des.c
+  libc-top-half/musl/src/crypt/crypt_md5.c
+  libc-top-half/musl/src/crypt/crypt_r.c
+  libc-top-half/musl/src/crypt/crypt_sha256.c
+  libc-top-half/musl/src/crypt/crypt_sha512.c
+  libc-top-half/musl/src/crypt/encrypt.c
+  libc-top-half/sources/arc4random.c)
+target_include_directories(libc-top-half PRIVATE
+  libc-top-half/musl/src/include
+  libc-top-half/musl/src/internal
+  libc-top-half/musl/arch/wasm32
+  libc-top-half/musl/arch/generic
+  libc-top-half/headers/private)
+target_compile_options(libc-top-half PRIVATE
+  -Wno-parentheses
+  -Wno-shift-op-parentheses
+  -Wno-bitwise-op-parentheses
+  -Wno-logical-op-parentheses
+  -Wno-string-plus-int
+  -Wno-dangling-else
+  -Wno-unknown-pragmas)
+add_dependencies(libc-top-half sysroot-headers)
+
+target_link_libraries(c PRIVATE
+  libc-bottom-half
+  libc-top-half
+  dlmalloc)
+
+add_custom_command(TARGET c POST_BUILD
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libm.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/librt.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libpthread.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libcrypt.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libutil.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libxnet.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libresolv.a
+  COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/libdl.a)
```

Comment:
With the Makefile-based system we hit a problem where we had too many files for a Windows commandline, and there's a manual workaround that splits up the `ar` command into multiple invocations. Is that no longer a problem?

Reply:
Right, that is an issue.  However, CMake is supposed to take responsibility for building up the command line appropriately.  There isnt supposed to be anything in the CMakeLists.txt to support that.  If that breaks, thats a bug in CMake, and we can deal with that separately, but it seems to build currently on Windows (as this changes the CI over to the CMake build).

Reply:
What is the advantage of supporting both at once?   It seems like a pain to support two built system without much benifit, and there is risk it will cause confusion.  I think I'd rather see this new build system replace the existing one atomically if/when it proves itself to be superior.

If you we do decide the land it alongside, then lets make it very clear that its experimental, and not make it part of the CI (to avoid blocking forward progress on maintaining dual build systems).